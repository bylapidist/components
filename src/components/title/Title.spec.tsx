import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Title, TitleBySize } from './index';

test('it works as h1', () => {
    const tree = renderer.create(<Title size={1}>H1</Title>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('it works as h2', () => {
    const tree = renderer.create(<Title size={2}>H2</Title>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('it works as h3', () => {
    const tree = renderer.create(<Title size={3}>H3</Title>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('it works as h4', () => {
    const tree = renderer.create(<Title size={4}>H4</Title>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('it works as h5', () => {
    const tree = renderer.create(<Title size={5}>H5</Title>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('it works as h6', () => {
    const tree = renderer.create(<Title size={6}>H6</Title>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('it maps to the correct title element', () => {
    const tree = renderer.create(<TitleBySize size={3}>H3</TitleBySize>);
    expect(tree).toMatchSnapshot();
    expect(tree.root.findByType('h3')).toBeTruthy();
});
