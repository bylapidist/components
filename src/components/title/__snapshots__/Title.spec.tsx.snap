// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works as h1 1`] = `
.c0 {
  font-size: 36px;
}

.c1 {
  font-family: 'Montserrat',sans-serif;
  display: block;
  width: 100%;
  margin: 0 0 4px;
  color: #292b3e;
  line-height: 1.6;
  -webkit-letter-spacing: 0.8px;
  -moz-letter-spacing: 0.8px;
  -ms-letter-spacing: 0.8px;
  letter-spacing: 0.8px;
}

<h1
  className="c0 c1"
>
  H1
</h1>
`;

exports[`it works as h2 1`] = `
.c0 {
  font-size: 24px;
}

.c1 {
  font-family: 'Montserrat',sans-serif;
  display: block;
  width: 100%;
  margin: 0 0 4px;
  color: #292b3e;
  line-height: 1.6;
  -webkit-letter-spacing: 0.8px;
  -moz-letter-spacing: 0.8px;
  -ms-letter-spacing: 0.8px;
  letter-spacing: 0.8px;
}

<h2
  className="c0 c1"
>
  H2
</h2>
`;

exports[`it works as h3 1`] = `
.c0 {
  font-size: 16px;
}

.c1 {
  font-family: 'Montserrat',sans-serif;
  display: block;
  width: 100%;
  margin: 0 0 4px;
  color: #292b3e;
  line-height: 1.6;
  -webkit-letter-spacing: 0.8px;
  -moz-letter-spacing: 0.8px;
  -ms-letter-spacing: 0.8px;
  letter-spacing: 0.8px;
}

<h3
  className="c0 c1"
>
  H3
</h3>
`;

exports[`it works as h4 1`] = `
.c0 {
  font-size: 14px;
}

.c1 {
  font-family: 'Montserrat',sans-serif;
  display: block;
  width: 100%;
  margin: 0 0 4px;
  color: #292b3e;
  line-height: 1.6;
  -webkit-letter-spacing: 0.8px;
  -moz-letter-spacing: 0.8px;
  -ms-letter-spacing: 0.8px;
  letter-spacing: 0.8px;
}

<h4
  className="c0 c1"
>
  H4
</h4>
`;

exports[`it works as h5 1`] = `
.c0 {
  font-size: 14px;
}

.c1 {
  font-family: 'Montserrat',sans-serif;
  display: block;
  width: 100%;
  margin: 0 0 4px;
  color: #292b3e;
  line-height: 1.6;
  -webkit-letter-spacing: 0.8px;
  -moz-letter-spacing: 0.8px;
  -ms-letter-spacing: 0.8px;
  letter-spacing: 0.8px;
}

<h5
  className="c0 c1"
>
  H5
</h5>
`;

exports[`it works as h6 1`] = `
.c0 {
  font-size: 14px;
}

.c1 {
  font-family: 'Montserrat',sans-serif;
  display: block;
  width: 100%;
  margin: 0 0 4px;
  color: #292b3e;
  line-height: 1.6;
  -webkit-letter-spacing: 0.8px;
  -moz-letter-spacing: 0.8px;
  -ms-letter-spacing: 0.8px;
  letter-spacing: 0.8px;
}

<h6
  className="c0 c1"
>
  H6
</h6>
`;
